packages/emptystate/src/react/index.js

```diff
@@ -113,34 +113,40 @@ Illustration.propTypes = {
 class EmptyState extends React.PureComponent {
   constructor(props) {
     super(props)
-
     this.renderSmallIfElementLessThan = 450
   }
 
-  renderContent(ctx) {
+  renderContent({ hasRenderedOnce, size }) {
     return (
-      <Context.Provider value={ctx}>
-        <div
-          {...styles.emptyState(this.props, ctx)}
-          {...filterReactProps(this.props)}
-        >
-          {this.props.illustration}
-          {this.props.heading}
-          {this.props.caption}
-          {this.props.actions}
-        </div>
-      </Context.Provider>
+      <WithTheme>
+        {({ themeName }) => {
+          const ctx = { hasRenderedOnce, size, themeName }
```

Comment:
yeah, this unexpected rewiring that wasn't required before is not my favorite...